operations/helm/charts/mimir-distributed/templates/kafka/kafka-statefulset.yaml

```diff
@@ -0,0 +1,175 @@
+{{- if and .Values.kafka.enabled (not .Values.federation_frontend.disableOtherComponents) -}}
+{{- with .Values.kafka -}}
+{{- $replicas := .replicas | default 1 -}}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "mimir.resourceName" (dict "ctx" $ "component" "kafka") }}
+  labels:
+    {{- include "mimir.labels" (dict "ctx" $ "component" "kafka") | nindent 4 }}
+  annotations:
+    {{- toYaml .podAnnotations | nindent 4 }}
+  namespace: {{ $.Release.Namespace | quote }}
+spec:
+  podManagementPolicy: Parallel
+  replicas: {{ $replicas }}
+  selector:
+    matchLabels:
+      {{- include "mimir.selectorLabels" (dict "ctx" $ "component" "kafka") | nindent 6 }}
+  serviceName: {{ include "mimir.resourceName" (dict "ctx" $ "component" "kafka") }}-headless
+  template:
+    metadata:
+      labels:
+        {{- include "mimir.podLabels" (dict "ctx" $ "component" "kafka") | nindent 8 }}
+      annotations:
+        {{- include "mimir.podAnnotations" (dict "ctx" $ "component" "kafka") | nindent 8 }}
+      namespace: {{ $.Release.Namespace | quote }}
+    spec:
+      serviceAccountName: {{ template "mimir.serviceAccountName" $ }}
+      {{- if .priorityClassName }}
+      priorityClassName: {{ .priorityClassName }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .securityContext | nindent 8 }}
+      {{- if .image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .image.pullSecrets }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      {{- with .nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      terminationGracePeriodSeconds: {{ .terminationGracePeriodSeconds }}
+      containers:
+        - name: kafka
+          image: "{{ .image.repository }}:{{ .image.tag }}"
+          imagePullPolicy: {{ .image.pullPolicy }}
+          ports:
+            - containerPort: 9092
+              name: kafka
+              protocol: TCP
+            - containerPort: 9093
+              name: controller
+              protocol: TCP
+          env:
+            # Kubernetes expands env variables in lexical order. So "_POD_NAME" (sic) here is so the name was expanded before KAFKA_ADVERTISED_LISTENERS is.
```

Comment:
what does sic mean? 🤔

Reply:
> what does sic mean? 🤔

It means that `_POD_NAME` (with the underscore in front) is not a typo, but indeed how it's expected here. I.e. if you rename it to `POD_NAME` the `KAFKA_ADVERTISED_LISTENERS` won't be resolved to a pod name, and Kafka will fail to start with 

```
Exception in thread "main" java.lang.IllegalArgumentException: Error creating broker listeners from
  'PLAINTEXT://$(POD_NAME).mimir-kafka-headless.test-mimir.svc.cluster.local.:9092': Unable to parse PLAINTEXT://$(POD_NAME).mimir-kafka-headless.test-mimir.svc.cluster.local.:9092
```